scripts/relup-base-vsns.escript

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env escript
+%% -*- mode: erlang; -*-
+
+-mode(compile).
+
+-define(RED, "\e[31m").
+-define(RESET, "\e[39m").
+
+main(["base-vsns", To0, VsnDB]) ->
+    {ok, [VsnMap]} = file:consult(VsnDB),
+    To = strip_pre_release(To0),
+    #{from_versions := Froms} = fetch_version(To, VsnMap),
+    lists:foreach(
+     fun(From) ->
+       io:format(user, "~s~n", [From])
+     end,
+     filter_froms(Froms)),
+    halt(0);
+main(["otp-vsn-for", Vsn0, VsnDB]) ->
+    {ok, [VsnMap]} = file:consult(VsnDB),
+    Vsn = strip_pre_release(Vsn0),
+    #{otp := OtpVsn} = fetch_version(Vsn, VsnMap),
+    io:format(user, "~s~n", [OtpVsn]),
+    halt(0);
+main(["insert-new-vsn", NewVsn0, BaseFromVsn0, OtpVsn0, VsnDB]) ->
+    {ok, [VsnMap]} = file:consult(VsnDB),
+    NewVsn = strip_pre_release(NewVsn0),
+    BaseFromVsn = strip_pre_release(BaseFromVsn0),
+    OtpVsn = list_to_binary(OtpVsn0),
+    case VsnMap of
+      #{NewVsn := _} ->
+          print_warning("Version ~s already in DB!~n", [NewVsn]),
+          halt(1);
+      #{BaseFromVsn := _} ->
+          ok;
+      _ ->
+          print_warning("Version ~s not found in DB!~n", [BaseFromVsn]),
+          halt(1)
+    end,
+    NewVsnMap = insert_new_vsn(VsnMap, NewVsn, OtpVsn, BaseFromVsn),
+    file:write_file(VsnDB, io_lib:format("%% -*- mode: erlang; -*-\n\n~p.", [NewVsnMap])),
+    halt(0);
+main(["check-vsn-db", NewVsn0, VsnDB]) ->
+    {ok, [VsnMap]} = file:consult(VsnDB),
+    NewVsn = strip_pre_release(NewVsn0),
+    case check_all_vsns_schema(VsnMap) of
+        [] -> ok;
+        Problems ->
+            print_warning("Invalid Version DB ~s!~n", [VsnDB]),
+            print_warning("Problems found:~n"),
+            lists:foreach(
+              fun(Problem) ->
+                print_warning("  ~p~n", [Problem])
+              end, Problems),
+            halt(1)
+    end,
+    case VsnMap of
+        #{NewVsn := _} ->
+            io:format(user, "ok~n", []),
+            halt(0);
+        _ ->
+            Candidates = find_insertion_candidates(NewVsn, VsnMap),
+            print_warning("Version ~s not found in the version DB!~n", [NewVsn]),
+            [] =/= Candidates
+                andalso print_warning("Candidates for to insert this version into:~n"),
+            lists:foreach(
+              fun(Vsn) ->
+                io:format(user, "  ~s~n", [Vsn])
+              end, Candidates),
+            print_warning(
+              "To insert this version automatically, run:~n"
+              "./scripts/relup-base-vsns insert-new-vsn NEW-VSN BASE-FROM-VSN NEW-OTP-VSN ~s~n"
+              "And commit the results.  Be sure to revise the changes.~n"
+              "Otherwise, edit the file manually~n",
+              [VsnDB]),
+            halt(1)
+    end.
```

Comment:
It would be nice to have some help message that explains the purpose of the script and how to use it.

Reply:
Indeed.  I'll follow up this PR with an usage / description message.